public/app/features/dashboard-scene/saving/SaveDashboardAsForm.tsx

```diff
@@ -86,8 +89,19 @@ export function SaveDashboardAsForm({ dashboard, changeInfo }: Props) {
         </Stack>
       </>
     );
+  };
+  function renderFooter(error?: Error) {
+    if (isNameExistsError(error)) {
+      return <NameAlreadyExistsError cancelButton={cancelButton} saveButton={saveButton} />;
+    }
+    return normalFooter(error);
   }
-
+  useEffect(() => {
+    if (Object.keys(contentSent).length === 0) {
+      return;
+    }
+    setIsRefreshed(contentSent.title !== formValues.title.trim() || contentSent.folder !== formValues.folder);
```

Comment:
you're comparing objects here (`contentSent.folder !== formValues.folder`) which will use reference equality so this won't work correctly. probably better to just store the `folderUid` in `contentSent`